[BUG] Project_BUILD_HEADERS_ONLY=ON results in an error

**Describe the bug**
Building with Project_BUILD_HEADERS_ONLY=ON results in an error.

`CMake Error at cmake/CompilerWarnings.cmake:93 (target_compile_options):
  target_compile_options may only set INTERFACE properties on INTERFACE
  targets`


**To Reproduce**
`cmake -DProject_BUILD_HEADERS_ONLY=ON`

**Expected behavior**
This should succeed.

--- a/cmake/CompilerWarnings.cmake
+++ b/cmake/CompilerWarnings.cmake
@@ -87,7 +87,7 @@ function(set_project_warnings project_name)
     message(AUTHOR_WARNING "No compiler warnings set for '${CMAKE_CXX_COMPILER_ID}' compiler.")
   endif()
 
-  if(BUILD_HEADERS_ONLY)
+  if(${PROJECT_NAME}_BUILD_HEADERS_ONLY)
         target_compile_options(${project_name} INTERFACE ${PROJECT_WARNINGS})
   else()
         target_compile_options(${project_name} PUBLIC ${PROJECT_WARNINGS})

